IPython/lib/inputhook.py

```diff
@@ -177,13 +197,17 @@ def clear_app_refs(self, gui=None):
             the app for that toolkit.  References are not held for gtk or tk
             as those toolkits don't have the notion of an app.
         """
+        warn("`clear_app_refs` is deprecated since IPython 5.0 and will be removed in future versions.",
```

Comment:
Rather than warning on individual methods, it may be more informative and less noisy to warn if the class is instantiated.